instrumentation/google.golang.org/grpc/otelgrpc/stats_handler.go

```diff
@@ -135,7 +139,11 @@ func (h *clientHandler) HandleConn(context.Context, stats.ConnStats) {
 
 func (c *config) handleRPC(ctx context.Context, rs stats.RPCStats, isServer bool) { // nolint: revive  // isServer is not a control flag.
 	span := trace.SpanFromContext(ctx)
-	gctx, _ := ctx.Value(gRPCContextKey{}).(*gRPCContext)
+	gctx, ok := ctx.Value(gRPCContextKey{}).(*gRPCContext)
+	if !ok || gctx == nil {
+		otel.Handle(errMissinggRPCContext)
+		return
+	}
 	var messageId int64
 	metricAttrs := make([]attribute.KeyValue, 0, len(gctx.metricAttrs)+1)
 	metricAttrs = append(metricAttrs, gctx.metricAttrs...)
```

Comment:
```suggestion
	var metricAttrs []attribute.KeyValue
	var messageId int64

	gctx, ok := ctx.Value(gRPCContextKey{}).(*gRPCContext)
	if ok && gctx != nil {
		metricAttrs = make([]attribute.KeyValue, 0, len(gctx.metricAttrs)+1)
		metricAttrs = append(metricAttrs, gctx.metricAttrs...)
	}
```

Reading the rest of the method it looks like there's already an assumption that `gctx` may be `nil`, so simply guarding the use of it to set `metricAttrs` may be the better approach than bailing entirely.

Reply:
This doesn't look universal.  It is nil checked for `InPayload` and `OutPayload`, but not on `End`.  We should probably make them all nil checked.

Reply:
Woops, missed your comment @MadVikingGod, but I don't think I have the context here, what are `InPayload`, `OutPayload` and `End`?

Reply:
@pellared got it in a different comment: https://github.com/open-telemetry/opentelemetry-go-contrib/pull/4825#discussion_r1456321716